contracts/VaultV3.vy

```diff
@@ -551,13 +551,88 @@ def _max_deposit(receiver: address) -> uint256:
 
 @view
 @internal
-def _max_redeem(owner: address) -> uint256:
-    return self.balance_of[owner]
+def _max_withdraw(
+    owner: address,
+    max_loss: uint256,
+    strategies: DynArray[address, MAX_QUEUE]
+) -> uint256:
+    """
+    @dev Returns the max amount of `asset` an `owner` can withdraw.
 
-@view
-@internal
-def _max_withdraw(owner: address) -> uint256:
-    return self._convert_to_assets(self.balance_of[owner], Rounding.ROUND_DOWN)
+    This will do a full simulation of the withdraw in order to determine
+    how much is currently liquid and if the `max_loss` would allow for the 
+    tx to not revert.
+
+    This will track any expected loss to check if the tx will revert, but
+    not account for it in the amount returned since it is unrealised and 
```

Comment:
nit but `unrealised` is british english.
```suggestion
    not account for it in the amount returned since it is unrealized and 
```

Reply:
Yes, but thats whats used everywhere including the function name.

So I fear we are stuck using it in the comments too.

Unless we want to change all of them